File: boj_codes/C++17/2292.cpp
// https://www.acmicpc.net/problem/2292
// First Written : 20210106
// Last Modified : 20210106

#include <cstdio>

int main(){
    int N;
    scanf("%d", &N);

    if(N == 1)
        printf("1");
    
    else{
        N--;
        int radius = 1;

        while(N > 0){
            N -= radius * 6;
            radius++;
        }

        printf("%d", radius);
    }


    return 0;
}

File: old_contents/boj_codes/Python3/2292.py
#https://www.acmicpc.net/problem/2292
# 20200426

import sys

def find_room(N):
    if N == 1:
        return 1

    else:
        N -= 1
        radius = 1

        while N > 0:
            N -= radius * 6
            radius += 1

        return radius

N = int(sys.stdin.readline().rstrip())
print(find_room(N))

